decidim-admin/app/commands/decidim/admin/impersonate_user.rb

```diff
@@ -20,31 +19,38 @@ def initialize(form, user)
       #
       # Returns nothing.
       def call
-        return broadcast(:invalid) unless user.managed? && authorization_valid?
+        return broadcast(:invalid) unless form.valid?
+
+        transaction do
+          user.save! unless user.persisted?
+
+          create_authorization
+
+          create_impersonation_log
+        end
 
-        create_impersonation_log
         enqueue_expire_job
 
         broadcast(:ok)
       end
 
       private
 
-      attr_reader :user, :form
+      attr_reader :form
 
-      def authorization_valid?
-        return false unless form.valid?
-        Decidim::Authorization.where(
-          user: user,
-          name: form.authorization.handler_name,
-          unique_id: form.authorization.unique_id
-        ).any?
+      def user
+        form.user
+      end
+
+      def create_authorization
+        Authorization.create_or_update_from(form.authorization)
       end
 
       def create_impersonation_log
         Decidim::ImpersonationLog.create!(
           admin: form.current_user,
           user: user,
+          reason: form.reason,
```

Comment:
When impersonating an existing user I think we should send an email to them to let them when and why an admin has impersonated them. Otherwise without an audit no one would know about it.

Reply:
It would be great to also notify the user when the impersonation has finished.

Reply:
Uff, that seems reasonable but this is a new spec for me and I have no more availability to work on this... :disappointed:

Reply:
Note though that the idea of impersonations (at least as I see them), is that the impersonated user is present during the impersonating. I mean, that's why we ask for authorization every time, right?

Reply:
We can do it in another PR @deivid-rodriguez, don't worry.

And yes, the user should be present during the impersonation, but and admin could know (or make a note) the user's data and impersonate them later.